String properties not working in Python3's klayout package

Hello Matthias. I think I might have found a bug related to #109 . I am trying to set a cell/layout property with a string name and a string value. But I noticed that they were not saving correctly. Then, I tried to save a property with name "123", but upon reading the gds file, it read as 123 (integer). The values were translated correctly.

Here's a sample of what I am talking about.
```
___________________________________________ test_metadata ____________________________________________

    def test_metadata():
        save_options = kdb.SaveLayoutOptions()
        save_options.gds2_write_file_properties = True
        save_options.gds2_write_cell_properties = True
        load_options = kdb.LoadLayoutOptions()
        load_options.properties_enabled = True
        layout = kdb.Layout()
        TOP = layout.create_cell("TOP")
        TOP.set_property("123", "string")
        layout.write("tests/tmp/test_metadata.gds", save_options)
        print(layout.has_prop_id(), TOP.has_prop_id())
        print([x for x in layout.properties(TOP.prop_id)])
        layout2 = kdb.Layout()
        layout2.read("tests/tmp/test_metadata.gds", load_options)
        TOP = layout2.top_cell()
        print(layout2.has_prop_id(), TOP.has_prop_id())
        print([x for x in layout2.properties(TOP.prop_id)])
>       assert TOP.property("123") == "test1"
E       AssertionError: assert None == 'test1'
E        +  where None = <built-in method property of Cell object at 0x1274728c0>('123')
E        +    where <built-in method property of Cell object at 0x1274728c0> = <klayout.dbcore.Cell object at 0x1274728c0>.property

tests/layout/test_metadata.py:24: AssertionError
---------------------------------------- Captured stdout call ----------------------------------------
False True
[['123', 'string']]
False True
[[123, 'string']]
====================================== short test summary info =======================================
FAILED tests/layout/test_metadata.py::test_metadata - AssertionError: assert None == 'test1'
```

--- a/src/db/db/gsiDeclDbCell.cc
+++ b/src/db/db/gsiDeclDbCell.cc
@@ -1809,6 +1809,7 @@ Class<db::Cell> decl_Cell ("db", "Cell",
     "If no property with that key exists, it will create one. Using that method is more "
     "convenient than creating a new property set with a new ID and assigning that properties ID.\n"
     "This method may change the properties ID. "
+    "Note: GDS only supports integer keys. OASIS supports numeric and string keys. "
     "\n"
     "This method has been introduced in version 0.23."
   ) + 
@@ -3821,6 +3822,7 @@ Class<db::Instance> decl_Instance ("db", "Instance",
     "If no property with that key exists, it will create one. Using that method is more "
     "convenient than creating a new property set with a new ID and assigning that properties ID.\n"
     "This method may change the properties ID. "
+    "Note: GDS only supports integer keys. OASIS supports numeric and string keys. "
     "Calling this method may invalidate any iterators. It should not be called inside a "
     "loop iterating over instances.\n"
     "\n"

--- a/src/db/db/gsiDeclDbLayout.cc
+++ b/src/db/db/gsiDeclDbLayout.cc
@@ -1038,12 +1038,13 @@ Class<db::Layout> decl_Layout ("db", "Layout",
     "This method has been introduced in version 0.24."
   ) + 
   gsi::method_ext ("set_property", &set_layout_property,
-    "@brief Set the user property with the given key to the given value\n"
+    "@brief Sets the user property with the given key to the given value\n"
     "@args key, value\n"
     "This method is a convenience method that sets the property with the given key to the given value. "
     "If no property with that key exists, it will create one. Using that method is more "
     "convenient than creating a new property set with a new ID and assigning that properties ID.\n"
     "This method may change the properties ID. "
+    "Note: GDS only supports integer keys. OASIS supports numeric and string keys. "
     "\n"
     "This method has been introduced in version 0.24."
   ) + 

--- a/src/db/db/gsiDeclDbShape.cc
+++ b/src/db/db/gsiDeclDbShape.cc
@@ -1292,6 +1292,7 @@ Class<db::Shape> decl_Shape ("db", "Shape",
     "If no property with that key exists, it will create one. Using that method is more "
     "convenient than creating a new property set with a new ID and assigning that properties ID.\n"
     "This method may change the properties ID. "
+    "Note: GDS only supports integer keys. OASIS supports numeric and string keys. "
     "Calling this method will invalidate any iterators. It should not be called inside a "
     "loop iterating over shapes.\n"
     "\n"

--- a/src/lay/lay/doc/programming/database_api.xml
+++ b/src/lay/lay/doc/programming/database_api.xml
@@ -473,6 +473,13 @@ shape.set_property(1, "NewValue")</pre>
   and "delete_property" also are provided for cells and cell instances (<class_doc href="Cell"/> and <class_doc href="Instance"/>).
   </p>
 
+  <p>
+  <b>Note:</b> The GDS format does not have string names for properties. As a result GDS only supports numeric property keys.
+  OASIS, on the other hand, can handle string and numeric property "names". When saving layouts in GDS format, KLayout tries
+  to convert the properties' names into numbers if possible (i.e. if it sees that the string is a number). If it can't, then
+  the property is not saved. When the layout is read by KLayout upon opening, it gets converted to integer.
+  </p>
+
   <h2>The LayerInfo class</h2>
 
   <keyword name="LayerInfo"/>

